src/main/kotlin/uk/gov/dluhc/printapi/database/repository/CertificateRepository.kt

```diff
@@ -31,3 +31,10 @@ interface CertificateRepository : JpaRepository<Certificate, UUID> {
     )
     fun getPrintRequestStatusCount(startInstant: Instant, endInstant: Instant, status: Status): Int
 }
+
+object CertificateRepositoryExtensions {
+    fun CertificateRepository.findByPrintRequestStatusAndBatchId(status: Status, batchId: String): List<Certificate> {
+        return findByPrintRequestsBatchId(batchId)
+            .filter { it.printRequests.stream().anyMatch { printRequest -> printRequest.getCurrentStatus().status == status } }
```

Comment:
Minor, @Neil-Massey. Seems like there is a slightly shorter version of this in Kotlin without using `.stream()`:

<pre>
    fun CertificateRepository.findByPrintRequestStatusAndBatchId(status: Status, batchId: String): List<Certificate> {
        return findByPrintRequestsBatchId(batchId)
            .filter { <b>it.printRequests.any {</b> printRequest -> printRequest.getCurrentStatus().status == status } }
    }
</pre>

Reply:
Done